src/nunit.analyzers/DisposeFieldsInTearDown/DisposeFieldsInTearDownAnalyzer.cs

```diff
@@ -0,0 +1,442 @@
+            id: AnalyzerIdentifiers.FieldIsNotDisposedInTearDown,
+            title: DisposeFieldsInTearDownConstants.FieldIsNotDisposedInTearDownTitle,
+            messageFormat: DisposeFieldsInTearDownConstants.FieldIsNotDisposedInTearDownMessageFormat,
+            category: Categories.Structure,
+            defaultSeverity: DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            description: DisposeFieldsInTearDownConstants.FieldIsNotDisposedInTearDownDescription);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(fieldIsNotDisposedInTearDown);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeDisposableFields, SyntaxKind.ClassDeclaration);
+        }
+
+        private static void AnalyzeDisposableFields(SyntaxNodeAnalysisContext context)
+        {
+            var classDeclaration = (ClassDeclarationSyntax)context.Node;
+
+            SemanticModel? model = context.SemanticModel;
+
+            var typeSymbol = model.GetDeclaredSymbol(classDeclaration, context.CancellationToken);
+            if (typeSymbol is null)
+            {
+                return;
+            }
+
+            if (typeSymbol.IsDisposable())
+            {
+                // If the type is Disposable, the MS Analyzer will conflict, so bail out.
+                return;
+            }
+
+            var fieldDeclarations = classDeclaration.Members
+                                                    .OfType<FieldDeclarationSyntax>()
+                                                    .Select(x => x.Declaration)
+                                                    .SelectMany(x => x.Variables);
+
+            Dictionary<string, VariableDeclaratorSyntax> fields = fieldDeclarations.ToDictionary(x => x.Identifier.Text);
+            HashSet<string> fieldNames = new HashSet<string>(fields.Keys);
+
+            ImmutableArray<ISymbol> members = typeSymbol.GetMembers();
+            var methods = members.OfType<IMethodSymbol>().Where(m => !m.IsStatic).ToArray();
+            var oneTimeTearDownMethods = methods.Where(m => HasAttribute(m, "OneTimeTearDownAttribute")).ToImmutableHashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
+            var oneTimeSetUpMethods = methods.Where(m => HasAttribute(m, "OneTimeSetUpAttribute")).ToImmutableHashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
+            var setUpMethods = methods.Where(m => HasAttribute(m, "SetUpAttribute")).ToImmutableHashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
+            var tearDownMethods = methods.Where(m => HasAttribute(m, "TearDownAttribute")).ToImmutableHashSet<IMethodSymbol>(SymbolEqualityComparer.Default);
```

Comment:
I'm wondering if we should have constants for these strings `"OneTimeTearDownAttribute"`, `"OneTimeSetUpAttribute"` ...., but I think we can do this in another (and later) PR

Reply:
Added constants. Will create separate PR for usages of these strings in other existing code.